File: RaftSwiftGRPC/Sources/RaftCore/Utils/Client/ServerStateResponse.swift
/// Node to Client, response to ServerStateRequest
public struct ServerStateResponse: Sendable {
    /// The ID of the server
    public let id: Int

    /// The state of the server
    public let state: ServerState

    /// Initializes a new ServerStateResponse
    /// - Parameters:
    ///   - id: The ID of the server
    ///   - state: The state of the server
    public init(id: Int, state: ServerState) {
        self.id = id
        self.state = state
    }
}

File: RaftKotlin/raft-node/src/main/kotlin/raft/types/Client/ServerStateResponse.kt
package raft.types.client

import raft.types.ServerState

/**
 * A response to a server state request.
 */
data class ServerStateResponse(
    /**
     * The ID of the server
     */
    val id: Int,
    /**
     * The state of the server
     */
    val state: ServerState
)
